src/puppetlabs/trapperkeeper/services/webserver/jetty10_core.clj

```diff
@@ -266,8 +266,14 @@
   [request-header-size]
   (let [http-config (doto (HttpConfiguration.)
                       (.setSendDateHeader true)
-                      (.setSendServerVersion false))]
-    (if request-header-size
+                      (.setSendServerVersion false)
+                      ;; LEGACY uri compliance mode that models Jetty-9.4 behavior by allowing
+                      ;; UriCompliance.Violation.AMBIGUOUS_PATH_SEGMENT, UriCompliance.Violation.AMBIGUOUS_EMPTY_SEGMENT,
+                      ;; UriCompliance.Violation.AMBIGUOUS_PATH_SEPARATOR, UriCompliance.Violation.AMBIGUOUS_PATH_ENCODING
+                      ;; and UriCompliance.Violation.UTF16_ENCODINGS
+                      ;; https://www.eclipse.org/jetty/javadoc/jetty-10/org/eclipse/jetty/http/UriCompliance.html#LEGACY
+                      (.setUriCompliance UriCompliance/LEGACY))]
```

Comment:
I initally started enumerating what we need here, but are there any of these we do not need?

Reply:
I think we only want to do this in the case that `normalize-request-uri` is "true" (which is a config option).  There are only some services that really need the URI normalization (Puppet Server comes to mind), and most of the time we want to reject these things as early as possible.

Reply:
And then we would need some tests that demonstrate the behavior if `normalize-request-uri` is true and false.

Reply:
Hmm I see, however `normalize-request-uri` isn't a config option, it's rather a option when the context handler is added https://github.com/rileynewton/trapperkeeper-webserver-jetty10/blob/PE-35977-resolve-uri-compliance-issue-in-jetty10-server/src/puppetlabs/trapperkeeper/services/webserver/jetty10_core.clj#L744 . It seems rather downstream of the creation of many objects, if we only want to ignore these then, it would seem we'd have to reconstruct a lot just for those requests unless there's some other mechanism to change what violations are permitted with an already running server/configuration.

Reply:
Drat.  Ok.   That makes sense.